controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java

```diff
@@ -415,24 +411,23 @@ private boolean isSuspendedInAnotherZone(Instance instance, ZoneId zone) {
     }
 
     /** Returns whether the given job can trigger at the given instant */
-    public boolean triggerAt(Instant instant, JobType job, Versions versions, Instance instance, DeploymentSpec deploymentSpec) {
-        Optional<JobStatus> jobStatus = instance.deploymentJobs().statusOf(job);
-        if (jobStatus.isEmpty()) return true;
-        if (jobStatus.get().pausedUntil().isPresent() && jobStatus.get().pausedUntil().getAsLong() > clock.instant().toEpochMilli()) return false;
-        if (jobStatus.get().isSuccess()) return true; // Success
-        if (jobStatus.get().lastCompleted().isEmpty()) return true; // Never completed
-        if (jobStatus.get().firstFailing().isEmpty()) return true; // Should not happen as firstFailing should be set for an unsuccessful job
-        if ( ! versions.targetsMatch(jobStatus.get().lastCompleted().get())) return true; // Always trigger as targets have changed
+    public boolean triggerAt(Instant instant, JobType job, JobStatus jobStatus, Versions versions, Instance instance, DeploymentSpec deploymentSpec) {
+        if (instance.deploymentJobs().statusOf(job).map(status -> status.pausedUntil().orElse(0)).orElse(0L) > clock.millis()) return false;
```

Comment:
Probably possible to `flatMap` the optionals to get rid of the duplicate `orElse`.

Reply:
No, it's an OptionalLong :'(